lib/inspect_rpmdeps.c

```diff
@@ -378,6 +406,70 @@ static bool check_explicit_lib_deps(struct rpminspect *ri, Header h, deprule_lis
                     found = true;
                 }
 
+                /* requires could be handled by a transitive dependency */
+                if (!found) {
+                    /*
+                     * Collect all Requires down the build/
+                     * This is weird, but first walk the current
+                     * package's Requires list and collect the package
+                     * names.  Then the next while loop will walk all
+                     * of those Requires and look at those peer
+                     * packages and gather their Requires until there
+                     * are no more to gather.  We have found it via a
+                     * transitive Requires if the transitive list
+                     * contains the potential_prov's package name.
+                     */
+                    TAILQ_FOREACH(verify, after_deps, items) {
+                        if (verify->type == TYPE_REQUIRES && *verify->requirement != '/'
+                            && !strprefix(verify->requirement, SHARED_LIB_PREFIX) && !strstr(verify->requirement, SHARED_LIB_SUFFIX)) {
+                            isareq = remove_isa_substring(verify->requirement);
+                            assert(isareq != NULL);
+
+                            if (is_subpackage(ri->peers, isareq) && !list_contains(transitive, isareq)) {
+                                transitive = list_add(transitive, isareq);
+                            }
+
+                            free(isareq);
+                        }
+                    }
+
+                    collect = true;
+
+                    while (collect) {
+                        TAILQ_FOREACH(peer, ri->peers, items) {
+                            tn = headerGetString(peer->after_hdr, RPMTAG_NAME);
+                            collect = false;
+
+                            TAILQ_FOREACH(entry, transitive, items) {
+                                if (!strcmp(tn, entry->data)) {
+                                    TAILQ_FOREACH(verify, peer->after_deprules, items) {
+                                        if (verify->type == TYPE_REQUIRES
+                                            && *verify->requirement != '/'
+                                            && !strprefix(verify->requirement, SHARED_LIB_PREFIX) && !strstr(verify->requirement, SHARED_LIB_SUFFIX)) {
+                                            isareq = remove_isa_substring(verify->requirement);
+                                            assert(isareq != NULL);
+
+                                            if (is_subpackage(ri->peers, isareq) && !list_contains(transitive, isareq)) {
+                                                transitive = list_add(transitive, isareq);
```

Comment:
Could you break out of the `TAILQ_FOREACH` here if the `pn` is found instead of continuing the breadth-first search?